lxi-tools 2.5 (new formula)

robcarruthers opened the pull request:
<!-- Use [x] to mark item done, or just click the checkboxes with device pointer -->

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

SMillerDev commented on Formula/lxi-tools.rb:
```diff
@@ -0,0 +1,31 @@
+  end
+
+  test do
+    system "#{bin}/lxi", "-v"
```
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

carlocab commented on Formula/lxi-tools.rb:
```diff
@@ -0,0 +1,31 @@
+  head "https://github.com/lxi-tools/lxi-tools.git", branch: "master"
+
+  depends_on "meson" => :build
+  depends_on "ninja" => :build
+
+  depends_on "desktop-file-utils"
```
```suggestion
  depends_on "ninja" => :build
  depends_on "pkg-config" => :build
  depends_on "desktop-file-utils"
```

chenrui333 commented:
`FileNotFoundError: [Errno 2] No such file or directory: 'gtk-update-icon-cache'`

robcarruthers commented:
We have fixed that error on master. so it should work if you install with --HEAD.

Its likely to be sometime before 2.6 is released and incorporate this change.

We assumed, because this formula enables a new install method for lxi-tools, in addition to installing from source, we could document the need for --HEAD in the interim and updated everything once 2.6 is released. Is this correct or could you recommend a better way for our users to install with homebrew now?

SMillerDev commented:
It won't be merged if it doesn't work. Additionally `--HEAD` installs are not supported by Homebrew. If it breaks you'll have to figure it out yourself.

robcarruthers commented:
Could you please help with some guidance, I am new to this?

The error has been [fixed on master](https://github.com/lxi-tools/lxi-tools/issues/72#issuecomment-1646782275), its just not released yet.

Since --HEAD is not supported, is there any way to make this formula work without waiting for a new release?

SMillerDev commented:
Since --HEAD is not supported, is there any way to make this formula work without waiting for a new release?

You can consider hosting it in a personal tap, which is very easy to do: https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap. Otherwise, Homebrew has managed without lxi-tools for a couple of years already, we can wait for the next release.
